src/librustc_typeck/check/mod.rs

```diff
@@ -944,10 +944,19 @@ impl<'a, 'gcx, 'tcx> Visitor<'gcx> for GatherLocalsVisitor<'a, 'gcx, 'tcx> {
     // Add explicitly-declared locals.
     fn visit_local(&mut self, local: &'gcx hir::Local) {
         let o_ty = match local.ty {
-            Some(ref ty) => Some(self.fcx.to_ty(&ty)),
-            None => None
+            Some(ref ty) => {
+                let o_ty = self.fcx.to_ty(&ty);
+
+                let (c_ty, _orig_values) = self.fcx.inh.infcx.canonicalize_query(&o_ty);
```

Comment:
So, I realize I guided you wrongly. I think you want to use `canonicalize_response`. These two functions differ in how they treat lifetimes -- in particular, `canonicalize_query` is designed for trait queries, and it replaces **all lifetimes** with variables -- including `'static`! That however is the opposite of what we want here. `canonicalize_response`, in contrast, will only replace lifetime *variables*, which is exactly what we want.

Reply:
Fixed this and it the test is passing again locally!